File: tools/demo/mock/run_demo.py
import time
from pathlib import Path

from playwright.sync_api import sync_playwright

SCREENSHOT_DIR = Path(__file__).resolve().parents[3] / "frontend" / "public" / "assets"


def run_demo():
    print("🚀 Starting True Full-Stack Mock Data Demo...")
    
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True, args=['--window-size=1440,900'])
        page = browser.new_page(viewport={'width': 1440, 'height': 900})
        
        try:
            print("1️⃣ Navigating to Local RAG Interface...")
            page.goto("http://localhost:3000", wait_until="networkidle")
            page.fill('input[autocomplete="username"]', "demo")
            page.fill('input[autocomplete="current-password"]', "demo")
            page.click("button >> text='Sign in'")
            page.wait_for_selector("textarea")
            page.screenshot(path=str(SCREENSHOT_DIR / "01_Initial_Interface.png"))
            print("   📸 Saved 01_Initial_Interface.png")
            
            print("2️⃣ Submitting vague medical query (API test)...")
            page.fill("textarea", "show me lab results")
            page.keyboard.press("Enter")
            
            time.sleep(2.0)
            page.screenshot(path=str(SCREENSHOT_DIR / "02_Clarification_Requested.png"))
            print("   📸 Saved 02_Clarification_Requested.png")
            
            print("3️⃣ Selecting a clarification option...")
            page.click("button >> text='PATTERSON'")
            
            time.sleep(2.5)
            page.screenshot(path=str(SCREENSHOT_DIR / "03_RBAC_Inline_Masking.png"))
            print("   📸 Saved 03_RBAC_Inline_Masking.png")
            
            print("4️⃣ Querying standard clinical protocol...")
            page.fill("textarea", "what is the heparin dosage?")
            page.keyboard.press("Enter")
            
            time.sleep(2.5)
            page.screenshot(path=str(SCREENSHOT_DIR / "04_Standard_Retrieval.png"))
            print("   📸 Saved 04_Standard_Retrieval.png")

            print("\n✅ True Full-Stack E2E Demo finished successfully!")
            
        except Exception as e:
            print(f"❌ Demo failed: {e}")
        finally:
            browser.close()

if __name__ == "__main__":
    run_demo()

File: tools/demo/browser/run_demo.js
const puppeteer = require('puppeteer');
const path = require('path');

const delay = (ms) => new Promise(resolve => setTimeout(resolve, ms));
const screenshotPath = (name) => path.resolve(
    __dirname,
    '..',
    '..',
    '..',
    'frontend',
    'public',
    'assets',
    name
);

(async () => {
    console.log("🚀 Starting Enterprise Healthcare RAG Demo Automation...");

    // Launch browser
    const browser = await puppeteer.launch({
        headless: 'new',
        args: ['--window-size=1440,900']
    });
    const page = await browser.newPage();
    await page.setViewport({ width: 1440, height: 900 });

    try {
        console.log("1️⃣ Navigating to Local RAG Interface...");
        await page.goto('http://localhost:3000', { waitUntil: 'networkidle0' });
        await page.type('input[autocomplete="username"]', 'demo');
        await page.type('input[autocomplete="current-password"]', 'demo');
        await page.click('button[type="submit"]');
        await page.waitForSelector('textarea');
        await page.screenshot({ path: screenshotPath('01_Initial_Interface.png') });
        console.log("   📸 Saved 01_Initial_Interface.png");

        console.log("2️⃣ Submitting vague medical query...");
        await page.focus('textarea');
        await page.keyboard.type('show me lab results');
        await page.keyboard.press('Enter');

        // Give the UI time to render the response.
        await delay(2000);
        await page.screenshot({ path: screenshotPath('02_Clarification_Requested.png') });
        console.log("   📸 Saved 02_Clarification_Requested.png (Shows dynamic inline buttons)");

        console.log("3️⃣ Selecting a clarification option...");
        // Find and click the button containing "PATTERSON" (Case insensitive for new uppercase design)
        const clicked = await page.evaluate(() => {
            const buttons = Array.from(document.querySelectorAll('button'));
            const pattersonBtn = buttons.find(b => b.textContent && b.textContent.toUpperCase().includes('PATTERSON'));
            if (pattersonBtn) {
                pattersonBtn.click();
                return true;
            }
            return false;
        });

        if (!clicked) {
            console.log("   ❌ Could not find the clarification button!");
        }

        // Wait for the selected clarification response and inline mask.
        await delay(2500);
        await page.screenshot({ path: screenshotPath('03_RBAC_Inline_Masking.png') });
        console.log("   📸 Saved 03_RBAC_Inline_Masking.png (Shows inline masking)");

        console.log("4️⃣ Querying standard clinical protocol...");
        await page.focus('textarea');
        await page.keyboard.type('what is the heparin dosage?');
        await page.keyboard.press('Enter');

        await delay(2500);
        await page.screenshot({ path: screenshotPath('04_Standard_Retrieval.png') });
        console.log("   📸 Saved 04_Standard_Retrieval.png (Shows High Confidence protocol source)");

        console.log("\n✅ Demo finished successfully! Screenshots updated in frontend/public/assets/");

    } catch (e) {
        console.error("Demo failed:", e);
    } finally {
        await browser.close();
    }
})();
